### DISCUSSION TITLE
Handling non-literal C-style escapes

### DISCUSSION THREAD

**Discussion by J-Kappes** [nushell/nushell#10627] [Q&A]
Assuming
```nu
let str = `foo\'bar`
```
is there a way to get `foo'bar` out of `str` analogously to the output of `"foo\'bar"`? `$"($str)"` doesn't work.

**Comment by fdncred** [comment]
Here are two ways.

No escape
```nushell
let str = "foo'bar"
$str
foo'bar
```
Allow escaping
```nushell
let str = "foo\'bar"
$str
foo'bar
```

**Reply by J-Kappes** [reply]
Neither of those applies to the assumed starting condition. I have a string variable (not a literal) which contains the backslash character followed by the apostrophe character.

**Reply by fdncred** [reply]
I didn't get that you wanted to keep the backslash. I thought you were trying to escape the quote.

Since you have a single quote in the middle, you have to use double quotes on the outside. Since double quotes automatically escape, you'd have to write it like this:
```nushell
let str = "foo\\'bar"
$str
foo\'bar
```

It looks like you can do this too, using backticks
```nushell
let str = echo `foo\'bar`
$str
foo\'bar
```

Let me know if I'm still missing your point.

**Reply by J-Kappes** [reply]
Yes, I want to know how to apply the same kind of c-style escapes as you get with double-quoted literals to non-literals. Assuming I have a non-literal variable string containing c-style backslash escape, how do I interpret those escapes in the same way that I could if that non-literal variable string were a literal.

Assuming I already have
```
let str = `foo\'bar`
```
how do I get `foo'bar` out of `str` ?

**Reply by fdncred** [reply]
The only way I know to have escapes interpreted is to have them in double quotes.
